packages/language-service/lib/utils.js

```diff
@@ -0,0 +1,270 @@
+/**
+ * @typedef {import('estree').Program} Program
+ * @typedef {import('mdast').Root} Root
+ * @typedef {import('typescript').IScriptSnapshot} IScriptSnapshot
+ * @typedef {import('typescript').TextSpan} TextSpan
+ * @typedef {import('unified').Processor<Root>} Processor
+ * @typedef {import('unist').Node} Node
+ * @typedef {import('unist').Parent} Parent
+ * @typedef {import('unist').Position} Position
+ *
+ * @typedef {[start: number, end: number]} OffsetRange
+ *
+ * @typedef {object} MDXShadow
+ * @property {(start?: number, end?: number) => string} getText
+ * Same as {@link IScriptSnapshot.getText}, except omitting start and end, returns the entire text.
+ * @property {(position: number) => number | undefined} getShadowPosition
+ * Map a position from the real MDX document to the JSX shadow document.
+ * @property {(shadowPosition: number) => number} getRealPosition
+ * Map a position from the shadow document to the real MDX document.
+ * @property {unknown} [error]
+ * This is defined if a parsing error has occurred.
+ * @property {Root} ast
+ * The markdown AST (mdast).
+ *
+ * @typedef {MDXShadow & IScriptSnapshot} MDXSnapshot
+ */
+
+import {visit} from 'unist-util-visit'
+
+const componentStart = `
+/**
+ * Render the MDX contents.
+ *
+ * @param {MDXContentProps} props The props that have been passed to the MDX component.
```

Comment:
```suggestion
 * @param {MDXContentProps} props
 *   The props that have been passed to the MDX component.
```

I like this style because it’s more clear where the name ends and description starts, and it allows for more clear line wrapping, because the description starts at col 5 or so instead of 50+

Reply:
Either style is fine with me. I wish Prettier could format this.